packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -335,6 +339,14 @@ class AnimationController extends Animation<double>
       final double range = upperBound - lowerBound;
       final double remainingFraction = range.isFinite ? (target - _value).abs() / range : 1.0;
       simulationDuration = this.duration * remainingFraction;
+    } else if (target < _value + _kAnimateToTolerance.distance && target > _value - _kAnimateToTolerance.distance) {
+      // We are basically at the target, don't animate.
+      simulationDuration = Duration.ZERO;
+      if (target != _value) {
+        _internalSetValue(target);
+        notifyListeners();
+        target = _value;
+      }
     }
     stop();
     if (simulationDuration == Duration.ZERO) {
```

Comment:
_internalSetValue does the setting of _status as well (as the code below). We might want to refactor this to be clearer about setting this only once.